src/backend/distributed/utils/statistics_collection.c

```diff
@@ -0,0 +1,157 @@
+/*-------------------------------------------------------------------------
+ *
+ * statistics_collection.c
+ *	  Anonymous reports and statistics collection.
+ *
+ * Copyright (c) 2017, Citus Data, Inc.
+ *
+ *-------------------------------------------------------------------------
+ */
+
+#if HAVE_LIBCURL
+
+#include "postgres.h"
+
+#include <curl/curl.h>
+#include <sys/utsname.h>
+
+#include "access/xact.h"
+#include "citus_version.h"
+#include "distributed/metadata_cache.h"
+#include "distributed/statistics_collection.h"
+#include "distributed/worker_manager.h"
+#include "lib/stringinfo.h"
+
+bool EnableStatisticsCollection = true; /* send basic usage statistics to Citus */
+
+static uint64_t NextPow2(uint64_t n);
+static uint64_t ClusterSize(List *distributedTableList);
+static bool SendHttpPostRequest(const char *url, const char *postFields, long timeout_seconds);
+
+/*
+ * CollectBasicUsageStatistics sends basic usage statistics to Citus servers.
+ * This includes Citus version, table count rounded to next power of 2, cluster
+ * size rounded to next power of 2, worker node count, and uname data.
+ */
+void
+CollectBasicUsageStatistics(void)
+{
+	List *distributedTables = NIL;
+	uint64_t roundedDistTableCount = 0;
+	uint64_t roundedClusterSize = 0;
+	uint32_t workerNodeCount = 0;
+	const long timeout_seconds = 5;
+	struct utsname unameData;
+	StringInfo fields = makeStringInfo();
+
+	StartTransactionCommand();
+	distributedTables = DistributedTableList();
```

Comment:
Again I'm a bit worried about allocations performed here. Maybe it would be best to create a temporary memory context during stats collection and then just clear it at the end?

Reply:
:+1: